xonsh/tools.py

```diff
@@ -427,9 +427,20 @@ def print_exception(msg=None):
     env = getattr(builtins, '__xonsh_env__', os.environ)
     if 'XONSH_SHOW_TRACEBACK' not in env:
         sys.stderr.write('xonsh: For full traceback set: '
-                         '$XONSH_SHOW_TRACEBACK = True\n')
-    if env.get('XONSH_SHOW_TRACEBACK', False):
-        traceback.print_exc()
+                         '$XONSH_SHOW_TRACEBACK = True or '
+                         '$XONSH_SHOW_TRACEBACK = \'<filename>\'\n')
+    # get env option for traceback and convert it to a tuple
+    traceback_opt = env.get('XONSH_SHOW_TRACEBACK', False)
+    show_trace, out_file = to_tb_tuple(traceback_opt)
+    if show_trace:
+        if not out_file:
+            # if the user has not specified a file, output to stderr
+            traceback.print_exc()
+        else:
+            # if the user has specified a file, open it in append mode
+            # and output traceback info there, as well as to stderr
+            traceback.print_exc()
+            traceback.print_exc(file = open(out_file, 'a'))
```

Comment:
We probably need a check here in case the user has given an invalid file name (e.g., in a directory that doesn't exist).

Might also be good to put this in a `with open(out_file, 'a') as f` block or similar, just to play it safer about closing this file.